files/rsyslog/65-viaq-formatting.conf

```diff
@@ -16,36 +16,47 @@ if strlen($!MESSAGE) > 0 then {
             unset $!originalmsg;
             unset $!unparsed-data;
         }
+        # NOTE: We don't currently use PRESERVE_JSON_LOG - we unconditionally delete
+        # the original field - this is in accordance with what fluentd does - it
+        # may preserve the original field in the parse_json plugin, but it will
+        # unconditionally remove the field in the viaq filter
     }
     # ensure that $!message is set and $!MESSAGE is unset
     # if rsyslog is case sensitive, then $!MESSAGE == $!message
     if $!MESSAGE == $!message then {
         set $.save_message = $!message;
         unset $!MESSAGE; # which also unsets $!message
         set $!message = $.save_message;
-        unset $.save_message;
```

Comment:
Why is this line removed? $.save_message is consumed in the line before?  (I thought "set" is a copy.)

Reply:
It is an optimization - we don't use `$.` variables in the final message - it might be "cheaper" for rsyslog to just remove all of them at the end of record processing, rather than having to delete each one of them separately